src/wp-includes/default-filters.php

```diff
@@ -306,7 +306,10 @@
 add_action( 'after_switch_theme', '_wp_sidebars_changed' );
 add_action( 'wp_print_styles', 'print_emoji_styles' );
 
-if ( isset( $_GET['replytocom'] ) ) {
+if (
+	isset( $_GET['replytocom'] ) ||
+	( isset( $_GET['unapproved'] ) && isset( $_GET['moderation-hash'] ) )
```

Comment:
Oof I'm sorry for the minor nitpick: I think we can simplify this to `isset( $_GET['replytocom']] ) || isset( $_GET['unapproved'], $_GET['moderation-hash'] )`, because `isset()` accepts any number of arguments, and returns true if _all_ of them are set. We don't have to call `isset()` multiple times.

Reply:
I was in two minds about blocking robots on these pages, so asked @jono-alderson from Yoast who frequently contributes SEO recommendations to the WordPress project for some advice.

I've been told the `canonical` meta tag pointing to the main post's page should be adequate for these pages so I'll revert the relevant commit. 

> I'm sorry for the minor nitpick

No need to be sorry, I'd forgotten `isset()` allowed this and given the number of sites using WordPress, getting the code to it's best possible state is important. :)